src/erl.rs

```diff
@@ -1380,3 +1486,163 @@ pub fn is_erlang_standard_library_module(name: &str) -> bool {
+                    .append(retrn)
+                    .append(")")
+            }
+        }
+    }
+}
+
+impl Type {
+    pub fn to_erlang_type_spec(&self) -> Document {
+        tracing::trace!("spec_for_type: {:?}", &self);
+        match self {
+            Self::Var { typ } => match &*Arc::as_ref(typ).borrow() {
+                TypeVar::Generic { .. } => "any()".to_doc(),
+                TypeVar::Link { typ } => typ.to_erlang_type_spec(),
+                TypeVar::Unbound { .. } => "any()".to_doc(),
+            },
+            Self::App {
+                name, module, args, ..
+            } => {
+                if module.is_empty() {
+                    // built-ins
+                    // TODO: more builtins?
+                    match name.as_ref() {
+                        "Int" => "integer()".to_doc(),
+                        "String" => "binary()".to_doc(),
+                        "Bool" => "boolean()".to_doc(),
+                        "Float" => "float()".to_doc(),
+                        "List" => {
+                            let arg0 = args[0].to_erlang_type_spec();
+                            "list(".to_doc().append(arg0).append(")")
+                        }
+                        "Result" => {
+                            let arg0 = args[0].to_erlang_type_spec();
+                            let arg1 = args[1].to_erlang_type_spec();
+                            "{ok, "
+                                .to_doc()
+                                .append(arg0)
+                                .append("} | {error, ")
+                                .append(arg1)
+                                .append("}")
+                        }
+                        name => name.to_snake_case().to_doc().append("()"),
+                    }
+                } else {
+                    let args = concat(
+                        args.iter()
+                            .map(|a| a.to_erlang_type_spec())
+                            .intersperse(", ".to_doc()),
+                    );
+                    let mod_name = concat(
+                        module
+                            .iter()
+                            .map(|m| m.to_snake_case().to_doc())
+                            .intersperse(".".to_doc()),
+                    );
+
+                    mod_name
+                        .append(":")
+                        .append(name.to_snake_case())
+                        .append("(")
+                        .append(args)
+                        .append(")")
+                }
+            }
+
+            Self::Fn {
+                args,  //: Vec<Arc<Type>>,
+                retrn, //: Arc<Type>,
+            } => {
+                let args = concat(
+                    args.iter()
+                        .map(|a| a.to_erlang_type_spec())
+                        .intersperse(", ".to_doc()),
+                );
+                let retrn = retrn.to_erlang_type_spec();
+                "fun(("
```

Comment:
Double `(`?

Reply:
afaict, type specs for functions look like:
```
       fun()                    %% any function
       | fun((...) -> Type)     %% any arity, returning Type
       | fun(() -> Type)
       | fun((TList) -> Type)
```
https://erlang.org/doc/reference_manual/typespec.html"

so for this code it looks like `fun((` `args` `) -> ` &nbsp;`return` `)`

Reply:
Oh wow I've gotten how to Erlang